src/coreclr/jit/codegenxarch.cpp

```diff
@@ -7077,7 +7077,7 @@ void CodeGen::genCompareInt(GenTree* treeNode)
     // The type cannot be larger than the machine word size
     assert(genTypeSize(type) <= genTypeSize(TYP_I_IMPL));
     // TYP_UINT and TYP_ULONG should not appear here, only small types can be unsigned
-    assert(!varTypeIsUnsigned(type) || varTypeIsSmall(type));
+    assert(!varTypeIsUnsigned(type) || varTypeIsSmall(type) || compiler->compIsaSupportedDebugOnly(InstructionSet_AVX512F));
```

Comment:
We shouldn't need to modify this case, the return type should still by `TYP_INT` or `TYP_LONG` under AVX512F.

Reply:
Whether its signed to unsigned or not should be tracked via additional metadata on the node, such as the `simdBaseType`

Reply:
You are right. Thanks for pointing this out. I added this initially because I saw some errors in ```float -> ubyte``` tests but then went back and fixed it in morph. This was supposed to be a temp fix. Removed it.